utils/lru.hh

```diff
@@ -13,12 +13,24 @@
 
 class evictable {
     friend class lru;
+    // For bookkeeping, we want the unlinking of evictables to be explicit.
+    // E.g. if the cache's internal data structure consists of multiple lists, we would
+    // like to know which list is an element being removed from.
+    // Therefore, we are using auto_unlink only to be able to call unlink() in the move constructor
+    // and we do NOT rely on automatic unlinking in _lru_link's destructor.
```

Comment:
Can we instead refactor those types that can be linked into the LRU to be non-movable instead? Maybe by moving internals into a field with stable storage (moving only reassigning ownership)?
Then, instead or in addition to this comment, we'd have the requirement for users to unlink in the type itself.

Reply:
They have to be movable (LSA moves them around in order to annoy @tgrabiec)

Reply:
Right, forgot about compaction.